Revert the following commits from Bug: https://trac.sagemath.org/ticket/31517
because we don't have ipywidgets 7 yet (to make the next two patches apply).

From 3d26006bae99898c192476bf31e799c2e9abb87e Mon Sep 17 00:00:00 2001
From: Kwankyu Lee <ekwankyu@gmail.com>
Date: Fri, 9 Apr 2021 14:27:38 +0900
Subject: Sage output MathJax3 compatible htmls

---
 src/doc/de/tutorial/latex.rst               |  14 +-
 src/doc/en/tutorial/latex.rst               |  14 +-
 src/doc/fr/tutorial/latex.rst               |  14 +-
 src/doc/ja/tutorial/latex.rst               |  14 +-
 src/doc/pt/tutorial/latex.rst               |  14 +-
 src/sage/interacts/test_jupyter.rst         | 303 ++++++++++++++--------------
 src/sage/interfaces/mathematica.py          |   4 +-
 src/sage/misc/html.py                       | 103 ++++------
 src/sage/misc/table.py                      |  91 +++++----
 src/sage/repl/rich_output/backend_base.py   |  12 +-
 src/sage/repl/rich_output/output_browser.py |  14 +-
 src/sage/repl/rich_output/pretty_print.py   |   4 +-
 12 files changed, 294 insertions(+), 307 deletions(-)

--- a/sage/src/doc/de/tutorial/latex.rst
+++ b/sage/src/doc/de/tutorial/latex.rst
@@ -86,13 +86,13 @@
     sage: var('z')
     z
     sage: mj(z^12)
-    <html>\[\newcommand{\Bold}[1]{\mathbf{#1}}z^{12}\]</html>
+    <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbf{#1}}z^{12}</script></html>
     sage: mj(QQ)
-    <html>\[\newcommand{\Bold}[1]{\mathbf{#1}}\Bold{Q}\]</html>
+    <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbf{#1}}\Bold{Q}</script></html>
     sage: mj(ZZ['x'])
-    <html>\[\newcommand{\Bold}[1]{\mathbf{#1}}\Bold{Z}[x]\]</html>
+    <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbf{#1}}\Bold{Z}[x]</script></html>
     sage: mj(integrate(z^4, z))
-    <html>\[\newcommand{\Bold}[1]{\mathbf{#1}}\frac{1}{5} \, z^{5}\]</html>
+    <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbf{#1}}\frac{1}{5} \, z^{5}</script></html>
 
 Grundlegende Nutzung
 ====================
@@ -206,10 +206,10 @@
     sage: from sage.misc.html import MathJax
     sage: mj=MathJax()
     sage: mj(QQ)
-    <html>\[\newcommand{\Bold}[1]{\mathbf{#1}}\Bold{Q}\]</html>
+    <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbf{#1}}\Bold{Q}</script></html>
     sage: latex.blackboard_bold(True)
     sage: mj(QQ)
-    <html>\[\newcommand{\Bold}[1]{\mathbb{#1}}\Bold{Q}\]</html>
+    <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbb{#1}}\Bold{Q}</script></html>
     sage: latex.blackboard_bold(False)
 
 Dank der Erweiterbarkeit von TeX können Sie selbst Makros und Pakete
@@ -228,7 +228,7 @@
     sage: from sage.misc.html import MathJax
     sage: mj=MathJax()
     sage: mj(x+y)
-    <html>\[\newcommand{\Bold}[1]{\mathbf{#1}}\newcommand{\foo}{bar}x + y\]</html>
+    <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbf{#1}}\newcommand{\foo}{bar}x + y</script></html>
 
 Zusätzliche Makros, die so hinzugefügt wurden, werden auch vom
 systemweiten TeX genutzt, wenn MathJax an seine Grenzen gestoßen ist.
--- a/sage/src/doc/en/tutorial/latex.rst
+++ b/sage/src/doc/en/tutorial/latex.rst
@@ -83,13 +83,13 @@
     sage: var('z')
     z
     sage: mj(z^12)
-    <html>\[\newcommand{\Bold}[1]{\mathbf{#1}}z^{12}\]</html>
+    <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbf{#1}}z^{12}</script></html>
     sage: mj(QQ)
-    <html>\[\newcommand{\Bold}[1]{\mathbf{#1}}\Bold{Q}\]</html>
+    <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbf{#1}}\Bold{Q}</script></html>
     sage: mj(ZZ['x'])
-    <html>\[\newcommand{\Bold}[1]{\mathbf{#1}}\Bold{Z}[x]\]</html>
+    <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbf{#1}}\Bold{Z}[x]</script></html>
     sage: mj(integrate(z^4, z))
-    <html>\[\newcommand{\Bold}[1]{\mathbf{#1}}\frac{1}{5} \, z^{5}\]</html>
+    <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbf{#1}}\frac{1}{5} \, z^{5}</script></html>
 
 Basic Use
 =========
@@ -194,10 +194,10 @@
     sage: from sage.misc.html import MathJax
     sage: mj=MathJax()
     sage: mj(QQ)
-    <html>\[\newcommand{\Bold}[1]{\mathbf{#1}}\Bold{Q}\]</html>
+    <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbf{#1}}\Bold{Q}</script></html>
     sage: latex.blackboard_bold(True)
     sage: mj(QQ)
-    <html>\[\newcommand{\Bold}[1]{\mathbb{#1}}\Bold{Q}\]</html>
+    <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbb{#1}}\Bold{Q}</script></html>
     sage: latex.blackboard_bold(False)
 
 It is possible to take advantage of the extensible nature of
@@ -217,7 +217,7 @@
     sage: from sage.misc.html import MathJax
     sage: mj=MathJax()
     sage: mj(x+y)
-    <html>\[\newcommand{\Bold}[1]{\mathbf{#1}}\newcommand{\foo}{bar}x + y\]</html>
+    <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbf{#1}}\newcommand{\foo}{bar}x + y</script></html>
 
 Additional macros added this way will also be used in the event
 that the system-wide version of TeX is called on
--- a/sage/src/doc/fr/tutorial/latex.rst
+++ b/sage/src/doc/fr/tutorial/latex.rst
@@ -83,13 +83,13 @@
     sage: var('z')
     z
     sage: mj(z^12)
-    <html>\[\newcommand{\Bold}[1]{\mathbf{#1}}z^{12}\]</html>
+    <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbf{#1}}z^{12}</script></html>
     sage: mj(QQ)
-    <html>\[\newcommand{\Bold}[1]{\mathbf{#1}}\Bold{Q}\]</html>
+    <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbf{#1}}\Bold{Q}</script></html>
     sage: mj(ZZ['x'])
-    <html>\[\newcommand{\Bold}[1]{\mathbf{#1}}\Bold{Z}[x]\]</html>
+    <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbf{#1}}\Bold{Z}[x]</script></html>
     sage: mj(integrate(z^4, z))
-    <html>\[\newcommand{\Bold}[1]{\mathbf{#1}}\frac{1}{5} \, z^{5}\]</html>
+    <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbf{#1}}\frac{1}{5} \, z^{5}</script></html>
 
 Utilisation de base
 ===================
@@ -197,10 +197,10 @@
     sage: from sage.misc.html import MathJax
     sage: mj=MathJax()
     sage: mj(QQ)
-    <html>\[\newcommand{\Bold}[1]{\mathbf{#1}}\Bold{Q}\]</html>
+    <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbf{#1}}\Bold{Q}</script></html>
     sage: latex.blackboard_bold(True)
     sage: mj(QQ)
-    <html>\[\newcommand{\Bold}[1]{\mathbb{#1}}\Bold{Q}\]</html>
+    <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbb{#1}}\Bold{Q}</script></html>
     sage: latex.blackboard_bold(False)
 
 On peut aussi définir de nouvelles macros TeX ou charger des packages
@@ -220,7 +220,7 @@
     sage: from sage.misc.html import MathJax
     sage: mj=MathJax()
     sage: mj(x+y)
-    <html>\[\newcommand{\Bold}[1]{\mathbf{#1}}\newcommand{\foo}{bar}x + y\]</html>
+    <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbf{#1}}\newcommand{\foo}{bar}x + y</script></html>
 
 Ces macros supplémentaires sont disponibles aussi quand Sage appelle TeX pour
 compiler un fragment de document trop gros pour MathJax. C'est la fonction
--- a/sage/src/doc/ja/tutorial/latex.rst
+++ b/sage/src/doc/ja/tutorial/latex.rst
@@ -65,13 +65,13 @@
     sage: var('z')
     z
     sage: mj(z^12)
-    <html>\[\newcommand{\Bold}[1]{\mathbf{#1}}z^{12}\]</html>
+    <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbf{#1}}z^{12}</script></html>
     sage: mj(QQ)
-    <html>\[\newcommand{\Bold}[1]{\mathbf{#1}}\Bold{Q}\]</html>
+    <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbf{#1}}\Bold{Q}</script></html>
     sage: mj(ZZ[x])
-    <html>\[\newcommand{\Bold}[1]{\mathbf{#1}}\Bold{Z}[x]\]</html>
+    <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbf{#1}}\Bold{Z}[x]</script></html>
     sage: mj(integrate(z^4, z))
-    <html>\[\newcommand{\Bold}[1]{\mathbf{#1}}\frac{1}{5} \, z^{5}\]</html>
+    <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbf{#1}}\frac{1}{5} \, z^{5}</script></html>
 
 
 
@@ -169,10 +169,10 @@
     sage: from sage.misc.html import MathJax
     sage: mj=MathJax()
     sage: mj(QQ)
-    <html>\[\newcommand{\Bold}[1]{\mathbf{#1}}\Bold{Q}\]</html>
+    <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbf{#1}}\Bold{Q}</script></html>
     sage: latex.blackboard_bold(True)
     sage: mj(QQ)
-    <html>\[\newcommand{\Bold}[1]{\mathbb{#1}}\Bold{Q}\]</html>
+    <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbb{#1}}\Bold{Q}</script></html>
     sage: latex.blackboard_bold(False)
 
 新しいマクロやパッケージなどを追加して，TeXの高い拡張性を利用することができる．
@@ -193,7 +193,7 @@
     sage: from sage.misc.html import MathJax
     sage: mj=MathJax()
     sage: mj(x+y)
-    <html>\[\newcommand{\Bold}[1]{\mathbf{#1}}\newcommand{\foo}{bar}x + y\]</html>
+    <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbf{#1}}\newcommand{\foo}{bar}x + y</script></html>
 
 
 以上のようなやり方で追加したマクロは，MathJaxでは対応しきれない大規模な処理が発生してシステム上のTeXが呼ばれるような場合にも使われる．
--- a/sage/src/doc/pt/tutorial/latex.rst
+++ b/sage/src/doc/pt/tutorial/latex.rst
@@ -91,13 +91,13 @@
     sage: var('z')
     z
     sage: js(z^12)
-    <html>\[\newcommand{\Bold}[1]{\mathbf{#1}}z^{12}\]</html>
+    <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbf{#1}}z^{12}</script></html>
     sage: js(QQ)
-    <html>\[\newcommand{\Bold}[1]{\mathbf{#1}}\Bold{Q}\]</html>
+    <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbf{#1}}\Bold{Q}</script></html>
     sage: js(ZZ[x])
-    <html>\[\newcommand{\Bold}[1]{\mathbf{#1}}\Bold{Z}[x]\]</html>
+    <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbf{#1}}\Bold{Z}[x]</script></html>
     sage: js(integrate(z^4, z))
-    <html>\[\newcommand{\Bold}[1]{\mathbf{#1}}\frac{1}{5} \, z^{5}\]</html>
+    <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbf{#1}}\frac{1}{5} \, z^{5}</script></html>
 
 Uso Básico
 ==========
@@ -207,11 +207,11 @@
     sage: from sage.misc.html import MathJax
     sage: js = MathJax()
     sage: js(QQ)
-    <html>\[\newcommand{\Bold}[1]{\mathbf{#1}}\Bold{Q}\]</html>
+    <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbf{#1}}\Bold{Q}</script></html>
 
     sage: latex.blackboard_bold(True)
     sage: js(QQ)
-    <html>\[\newcommand{\Bold}[1]{\mathbb{#1}}\Bold{Q}\]</html>
+    <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbb{#1}}\Bold{Q}</script></html>
     sage: latex.blackboard_bold(False)
 
 É possível aproveitar os recursos do TeX adicionando novas funções
@@ -231,7 +231,7 @@
     sage: from sage.misc.html import MathJax
     sage: js = MathJax()
     sage: js(x+y)
-    <html>\[\newcommand{\Bold}[1]{\mathbf{#1}}\newcommand{\foo}{bar}x + y\]</html>
+    <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbf{#1}}\newcommand{\foo}{bar}x + y</script></html>
 
 Macros adicionais usadas dessa forma serão também usadas eventualmente
 se a versão do TeX no seu sistema for usada para lidar com algo muito
--- a/sage/src/sage/interacts/test_jupyter.rst
+++ b/sage/src/sage/interacts/test_jupyter.rst
@@ -39,107 +39,107 @@
 
     sage: test(interacts.algebra.polar_prime_spiral)  # long time
     Interactive function <function polar_prime_spiral at ...> with 6 widgets
-      interval: IntRangeSlider(value=(1, 1000), description='range', max=4000, min=1, step=10)
-      show_factors: Checkbox(value=True, description='show_factors')
-      highlight_primes: Checkbox(value=True, description='highlight_primes')
-      show_curves: Checkbox(value=True, description='show_curves')
-      n: IntSlider(value=89, description='number $n$', max=200, min=1)
-      dpi: IntSlider(value=100, description='dpi', max=300, min=10, step=10)
+      interval: IntRangeSlider(value=(1, 1000), description=u'range', max=4000, min=1, step=10)
+      show_factors: Checkbox(value=True, description=u'show_factors')
+      highlight_primes: Checkbox(value=True, description=u'highlight_primes')
+      show_curves: Checkbox(value=True, description=u'show_curves')
+      n: IntSlider(value=89, description=u'number $n$', max=200, min=1)
+      dpi: IntSlider(value=100, description=u'dpi', max=300, min=10, step=10)
     <h2>Polar Prime Spiral</h2>           <div style="white-space: normal;">          For more information about the factors in the spiral, visit           <a href="http://www.dcs.gla.ac.uk/~jhw/spirals/index.html" target="_blank">          Number Spirals by John Williamson</a>.</div>
-    \(n = 89\)
-    Pink Curve: \(n^2 + 8\)
-    Green Curve: \(n^2 + n + -1\)
+    <script type="math/tex">n = 89</script>
+    Pink Curve:  <script type="math/tex">n^2 + 8</script>
+    Green Curve: <script type="math/tex">n^2 + n + -1</script>
 
     sage: test(interacts.calculus.taylor_polynomial)
     Interactive function <function taylor_polynomial at ...> with 3 widgets
-      title: HTMLText(value='<h2>Taylor polynomial</h2>')
-      f: EvalText(value='e^(-x)*sin(x)', description='$f(x)=$', layout=Layout(max_width='81em'))
-      order: SelectionSlider(description='order', options=(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12), value=1)
-    \(f(x)\;=\;e^{\left(-x\right)} \sin\left(x\right)\)
-    \(\hat{f}(x;0)\;=\;x+\mathcal{O}(x^{2})\)
+      title: HTMLText(value=u'<h2>Taylor polynomial</h2>')
+      f: EvalText(value=u'e^(-x)*sin(x)', description=u'$f(x)=$', layout=Layout(max_width=u'81em'))
+      order: SelectionSlider(description=u'order', options=(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12), value=1)
+    <script type="math/tex">f(x)\;=\;e^{\left(-x\right)} \sin\left(x\right)</script>
+    <script type="math/tex">\hat{f}(x;0)\;=\;x+\mathcal{O}(x^{2})</script>
 
     sage: test(interacts.calculus.definite_integral)
     Interactive function <function definite_integral at ...> with 6 widgets
-      title: HTMLText(value='<h2>Definite integral</h2>')
-      f: EvalText(value='3*x', description='$f(x)=$', layout=Layout(max_width='81em'))
-      g: EvalText(value='x^2', description='$g(x)=$', layout=Layout(max_width='81em'))
-      interval: IntRangeSlider(value=(0, 3), description='Interval', max=10, min=-10)
-      x_range: IntRangeSlider(value=(0, 3), description='plot range (x)', max=10, min=-10)
-      selection: Dropdown(description='Select', index=2, options=('f', 'g', 'f and g', 'f - g'), value='f and g')
-    \(\int_{0.00}^{3.00}(\color{Blue}{f(x)})\,\mathrm{d}x=\int_{0.00}^{3.00}(3 \, x)\,\mathrm{d}x=13.50\)<br/>\(\int_{0.00}^{3.00}(\color{Green}{g(x)})\,\mathrm{d}x=\int_{0.00}^{3.00}(x^{2})\,\mathrm{d}x=9.00\)
+      title: HTMLText(value=u'<h2>Definite integral</h2>')
+      f: EvalText(value=u'3*x', description=u'$f(x)=$', layout=Layout(max_width=u'81em'))
+      g: EvalText(value=u'x^2', description=u'$g(x)=$', layout=Layout(max_width=u'81em'))
+      interval: IntRangeSlider(value=(0, 3), description=u'Interval', max=10, min=-10)
+      x_range: IntRangeSlider(value=(0, 3), description=u'plot range (x)', max=10, min=-10)
+      selection: Dropdown(description=u'Select', index=2, options=('f', 'g', 'f and g', 'f - g'), value='f and g')
+    <script type="math/tex">\int_{0.00}^{3.00}(\color{Blue}{f(x)})\,\mathrm{d}x=\int_{0.00}^{3.00}(3 \, x)\,\mathrm{d}x=13.50</script><br/><script type="math/tex">\int_{0.00}^{3.00}(\color{Green}{g(x)})\,\mathrm{d}x=\int_{0.00}^{3.00}(x^{2})\,\mathrm{d}x=9.00</script>
 
     sage: test(interacts.calculus.function_derivative)
     Interactive function <function function_derivative at ...> with 4 widgets
-      title: HTMLText(value='<h2>Derivative grapher</h2>')
-      function: EvalText(value='x^5-3*x^3+1', description='Function:', layout=Layout(max_width='81em'))
-      x_range: FloatRangeSlider(value=(-2.0, 2.0), description='Range (x)', max=15.0, min=-15.0)
-      y_range: FloatRangeSlider(value=(-8.0, 6.0), description='Range (y)', max=15.0, min=-15.0)
-    <center>\(\color{Blue}{f(x) = x^{5} - 3 \, x^{3} + 1}\)</center>
-    <center>\(\color{Green}{f'(x) = 5 \, x^{4} - 9 \, x^{2}}\)</center>
-    <center>\(\color{Red}{f''(x) = 20 \, x^{3} - 18 \, x}\)</center>
+      title: HTMLText(value=u'<h2>Derivative grapher</h2>')
+      function: EvalText(value=u'x^5-3*x^3+1', description=u'Function:', layout=Layout(max_width=u'81em'))
+      x_range: FloatRangeSlider(value=(-2.0, 2.0), description=u'Range (x)', max=15.0, min=-15.0)
+      y_range: FloatRangeSlider(value=(-8.0, 6.0), description=u'Range (y)', max=15.0, min=-15.0)
+    <center><script type="math/tex">\color{Blue}{f(x) = x^{5} - 3 \, x^{3} + 1}</script></center>
+    <center><script type="math/tex">\color{Green}{f'(x) = 5 \, x^{4} - 9 \, x^{2}}</script></center>
+    <center><script type="math/tex">\color{Red}{f''(x) = 20 \, x^{3} - 18 \, x}</script></center>
 
     sage: test(interacts.calculus.difference_quotient)
     Interactive function <function difference_quotient at ...> with 5 widgets
-      title: HTMLText(value='<h2>Difference quotient</h2>')
-      f: EvalText(value='sin(x)', description='f(x)', layout=Layout(max_width='81em'))
-      interval: FloatRangeSlider(value=(0.0, 10.0), description='Range', max=10.0)
-      a: IntSlider(value=5, description='$a$', max=10)
-      x0: IntSlider(value=2, description='$x_0$ (start point)', max=10)
+      title: HTMLText(value=u'<h2>Difference quotient</h2>')
+      f: EvalText(value=u'sin(x)', description=u'f(x)', layout=Layout(max_width=u'81em'))
+      interval: FloatRangeSlider(value=(0.0, 10.0), description=u'Range', max=10.0)
+      a: IntSlider(value=5, description=u'$a$', max=10)
+      x0: IntSlider(value=2, description=u'$x_0$ (start point)', max=10)
     <h2>Difference Quotient</h2>
     <div style="white-space: normal;">         <a href="https://en.wikipedia.org/wiki/Difference_quotient" target="_blank">         Wikipedia article about difference quotient</a></div>
     <h2>Difference Quotient</h2>
-    <br>\(\text{Line's equation:}\)
-    \(y = 1/3*(x - 5)*(sin(5) - sin(2)) + sin(5)\)<br>
-    \(\text{Slope:}\)
-    \(k = \frac{f(x_0)-f(a)}{x_0-a} = -0.62274\)<br>
+    <br><script type="math/tex">\text{Line's equation:}</script>
+    <script type="math/tex">y = 1/3*(x - 5)*(sin(5) - sin(2)) + sin(5)</script><br>
+    <script type="math/tex">\text{Slope:}</script>
+    <script type="math/tex">k = \frac{f(x_0)-f(a)}{x_0-a} = -0.62274</script><br>
 
     sage: test(interacts.calculus.quadratic_equation)
     Interactive function <function quadratic_equation at ...> with 3 widgets
-      A: IntSlider(value=1, description='A', max=7, min=-7)
-      B: IntSlider(value=1, description='B', max=7, min=-7)
-      C: IntSlider(value=-2, description='C', max=7, min=-7)
+      A: IntSlider(value=1, description=u'A', max=7, min=-7)
+      B: IntSlider(value=1, description=u'B', max=7, min=-7)
+      C: IntSlider(value=-2, description=u'C', max=7, min=-7)
     <h2>The Solutions of the Quadratic Equation</h2>
-    \(x^2 + x - 2 = 0\)
-    \(Ax^2 + Bx + C = 0\)
-    \(x = \frac{-B\pm\sqrt{B^2-4AC}}{2A} = \frac{-1\pm\sqrt{1^2-4*1*-2}}{2*1} = \frac{-1\pm\sqrt{\color{Green}{9}}}{2} = \begin{cases}1\\-2\end{cases}\)
+    <script type="math/tex">x^2 + x - 2 = 0</script>
+    <script type="math/tex">Ax^2 + Bx + C = 0</script>
+    <script type="math/tex">x = \frac{-B\pm\sqrt{B^2-4AC}}{2A} = \frac{-1\pm\sqrt{1^2-4*1*-2}}{2*1} = \frac{-1\pm\sqrt{\color{Green}{9}}}{2} = \begin{cases}1\\-2\end{cases}</script>
 
     sage: test(interacts.calculus.secant_method)
     Interactive function <function secant_method at ...> with 5 widgets
-      title: HTMLText(value='<h2>Secant method for numerical root finding</h2>')
-      f: EvalText(value='x^2-2', description='f(x)', layout=Layout(max_width='81em'))
-      interval: IntRangeSlider(value=(0, 4), description='range', max=5, min=-5)
-      d: IntSlider(value=3, description='10^-d precision', max=16, min=1)
-      maxn: IntSlider(value=10, description='max iterations', max=15)
-    \(\text{Precision }h = 10^{-d}=10^{-3}=0.00100\)
-    \({c = }1.4144038097709382\)
-    \({f(c) = }0.0005381370945443109\)
-    \(6 \text{ iterations}\)
+      title: HTMLText(value=u'<h2>Secant method for numerical root finding</h2>')
+      f: EvalText(value=u'x^2-2', description=u'f(x)', layout=Layout(max_width=u'81em'))
+      interval: IntRangeSlider(value=(0, 4), description=u'range', max=5, min=-5)
+      d: IntSlider(value=3, description=u'10^-d precision', max=16, min=1)
+      maxn: IntSlider(value=10, description=u'max iterations', max=15)
+    <script type="math/tex">\text{Precision }h = 10^{-d}=10^{-3}=0.00100</script>
+    <script type="math/tex">{c = }1.4144038097709382</script>
+    <script type="math/tex">{f(c) = }0.0005381370945443109</script>
+    <script type="math/tex">6 \text{ iterations}</script>
 
     sage: test(interacts.calculus.newton_method)
     Interactive function <function newton_method at ...> with 7 widgets
-      title: HTMLText(value='<h2>Newton method</h2>')
-      f: EvalText(value='x^2 - 2', description='f', layout=Layout(max_width='81em'))
-      c: IntSlider(value=6, description='Start ($x$)', max=10, min=-10)
-      d: IntSlider(value=3, description='$10^{-d}$ precision', max=16, min=1)
-      maxn: IntSlider(value=10, description='max iterations', max=15)
-      interval: IntRangeSlider(value=(0, 6), description='Interval', max=10, min=-10)
-      list_steps: Checkbox(value=False, description='List steps')
-    \(\text{Precision } 2h = 0.001\)
-    \({c = }1.4142141576301823\)
-    \({f(c) = }1.6836416460996873 \times 10^{-06}\)
-    \(6 \text{ iterations}\)
+      title: HTMLText(value=u'<h2>Newton method</h2>')
+      f: EvalText(value=u'x^2 - 2', description=u'f', layout=Layout(max_width=u'81em'))
+      c: IntSlider(value=6, description=u'Start ($x$)', max=10, min=-10)
+      d: IntSlider(value=3, description=u'$10^{-d}$ precision', max=16, min=1)
+      maxn: IntSlider(value=10, description=u'max iterations', max=15)
+      interval: IntRangeSlider(value=(0, 6), description=u'Interval', max=10, min=-10)
+      list_steps: Checkbox(value=False, description=u'List steps')
+    <script type="math/tex">\text{Precision } 2h = 0.001</script>
+    <script type="math/tex">{c = }1.4142141576301823</script>
+    <script type="math/tex">{f(c) = }1.6836416460996873 \times 10^{-06}</script>
+    <script type="math/tex">6 \text{ iterations}</script>
 
     sage: test(interacts.calculus.trapezoid_integration)
     Interactive function <function trapezoid_integration at ...> with 7 widgets
-      title: HTMLText(value='<h2>Trapezoid integration</h2>')
-      f: EvalText(value='x^2-5*x + 10', description='$f(x)=$', layout=Layout(max_width='81em'))
-      n: IntSlider(value=5, description='# divisions', min=1)
-      interval_input: ToggleButtons(description='Integration interval', options=('from slider', 'from keyboard'), value='from slider')
-      interval_s: IntRangeSlider(value=(0, 8), description='slider: ', max=10, min=-10)
-      interval_g: Grid(value=[[0, 8]], children=(Label(value='keyboard: '), VBox(children=(EvalText(value='0', layout=Layout(max_width='5em')),)), VBox(children=(EvalText(value='8', layout=Layout(max_width='5em')),))))
-      output_form: ToggleButtons(description='Computations form', options=('traditional', 'table', 'none'), value='traditional')
-    Function \(f(x)=x^{2} - 5 \, x + 10\)
-    Integral value to seven decimal places is: \(\displaystyle\int_{0.00}^{8.00} {f(x) \, \mathrm{d}x} = 90.666667\)
+      title: HTMLText(value=u'<h2>Trapezoid integration</h2>')
+      f: EvalText(value=u'x^2-5*x + 10', description=u'$f(x)=$', layout=Layout(max_width=u'81em'))
+      n: IntSlider(value=5, description=u'# divisions', min=1)
+      interval_input: ToggleButtons(description=u'Integration interval', options=('from slider', 'from keyboard'), value='from slider')
+      interval_s: IntRangeSlider(value=(0, 8), description=u'slider: ', max=10, min=-10)
+      interval_g: Grid(value=[[0, 8]], children=(Label(value=u'keyboard: '), VBox(children=(EvalText(value=u'0', layout=Layout(max_width=u'5em')),)), VBox(children=(EvalText(value=u'8', layout=Layout(max_width=u'5em')),))))
+      output_form: ToggleButtons(description=u'Computations form', options=('traditional', 'table', 'none'), value='traditional')
+    Function <script type="math/tex">f(x)=x^{2} - 5 \, x + 10</script>
+    Integral value to seven decimal places is: <script type="math/tex">\displaystyle\int_{0.00}^{8.00} {f(x) \, \mathrm{d}x} = 90.666667</script>
     <BLANKLINE>
                 <div class="math">
                 \begin{align*}
@@ -154,15 +154,15 @@
 
     sage: test(interacts.calculus.simpson_integration)
     Interactive function <function simpson_integration at ...> with 7 widgets
-      title: HTMLText(value='<h2>Simpson integration</h2>')
-      f: EvalText(value='x*sin(x)+x+1', description='$f(x)=$', layout=Layout(max_width='81em'))
-      n: IntSlider(value=6, description='# divisions', min=2, step=2)
-      interval_input: ToggleButtons(description='Integration interval', options=('from slider', 'from keyboard'), value='from slider')
-      interval_s: IntRangeSlider(value=(0, 10), description='slider: ', max=10, min=-10)
-      interval_g: Grid(value=[[0, 10]], children=(Label(value='keyboard: '), VBox(children=(EvalText(value='0', layout=Layout(max_width='5em')),)), VBox(children=(EvalText(value='10', layout=Layout(max_width='5em')),))))
-      output_form: ToggleButtons(description='Computations form', options=('traditional', 'table', 'none'), value='traditional')
-    Function \(f(x)=x \sin\left(x\right) + x + 1\)
-    Integral value to seven decimal places is: \(\displaystyle\int_{0.00}^{10.00} {f(x) \, \mathrm{d}x} = 67.846694\)
+      title: HTMLText(value=u'<h2>Simpson integration</h2>')
+      f: EvalText(value=u'x*sin(x)+x+1', description=u'$f(x)=$', layout=Layout(max_width=u'81em'))
+      n: IntSlider(value=6, description=u'# divisions', min=2, step=2)
+      interval_input: ToggleButtons(description=u'Integration interval', options=('from slider', 'from keyboard'), value='from slider')
+      interval_s: IntRangeSlider(value=(0, 10), description=u'slider: ', max=10, min=-10)
+      interval_g: Grid(value=[[0, 10]], children=(Label(value=u'keyboard: '), VBox(children=(EvalText(value=u'0', layout=Layout(max_width=u'5em')),)), VBox(children=(EvalText(value=u'10', layout=Layout(max_width=u'5em')),))))
+      output_form: ToggleButtons(description=u'Computations form', options=('traditional', 'table', 'none'), value='traditional')
+    Function <script type="math/tex">f(x)=x \sin\left(x\right) + x + 1</script>
+    Integral value to seven decimal places is: <script type="math/tex">\displaystyle\int_{0.00}^{10.00} {f(x) \, \mathrm{d}x} = 67.846694</script>
     <BLANKLINE>
             <div class="math">
             \begin{align*}
@@ -177,109 +177,110 @@
 
     sage: test(interacts.calculus.bisection_method)
     Interactive function <function bisection_method at ...> with 5 widgets
-      title: HTMLText(value='<h2>Bisection method</h2>')
-      f: EvalText(value='x^2-2', description='f(x)', layout=Layout(max_width='81em'))
-      interval: IntRangeSlider(value=(0, 4), description='range', max=5, min=-5)
-      d: IntSlider(value=3, description='$10^{-d}$ precision', max=8, min=1)
-      maxn: IntSlider(value=10, description='max iterations', max=50)
-    \(\text{Precision }h = 10^{-d}=10^{-3}=0.00100\)
-    \({c = }1.4140625\)
-    \({f(c) = }-0.00042724609375\)
-    \(9 \text{ iterations}\)
+      title: HTMLText(value=u'<h2>Bisection method</h2>')
+      f: EvalText(value=u'x^2-2', description=u'f(x)', layout=Layout(max_width=u'81em'))
+      interval: IntRangeSlider(value=(0, 4), description=u'range', max=5, min=-5)
+      d: IntSlider(value=3, description=u'$10^{-d}$ precision', max=8, min=1)
+      maxn: IntSlider(value=10, description=u'max iterations', max=50)
+    <script type="math/tex">\text{Precision }h = 10^{-d}=10^{-3}=0.00100</script>
+    <script type="math/tex">{c = }1.4140625</script>
+    <script type="math/tex">{f(c) = }-0.00042724609375</script>
+    <script type="math/tex">9 \text{ iterations}</script>
 
     sage: test(interacts.calculus.riemann_sum)
     Manual interactive function <function riemann_sum at ...> with 9 widgets
-      title: HTMLText(value='<h2>Riemann integral with random sampling</h2>')
-      f: EvalText(value='x^2+1', description='$f(x)=$', layout=Layout(max_width='41em'))
-      n: IntSlider(value=5, description='# divisions', max=30, min=1)
-      hr1: HTMLText(value='<hr>')
-      interval_input: ToggleButtons(description='Integration interval', options=('from slider', 'from keyboard'), value='from slider')
-      interval_s: IntRangeSlider(value=(0, 2), description='slider: ', max=10, min=-5)
-      interval_g: Grid(value=[[0, 2]], children=(Label(value='keyboard: '), VBox(children=(EvalText(value='0', layout=Layout(max_width='5em')),)), VBox(children=(EvalText(value='2', layout=Layout(max_width='5em')),))))
-      hr2: HTMLText(value='<hr>')
-      list_table: Checkbox(value=False, description='List table')
+      title: HTMLText(value=u'<h2>Riemann integral with random sampling</h2>')
+      f: EvalText(value=u'x^2+1', description=u'$f(x)=$', layout=Layout(max_width=u'41em'))
+      n: IntSlider(value=5, description=u'# divisions', max=30, min=1)
+      hr1: HTMLText(value=u'<hr>')
+      interval_input: ToggleButtons(description=u'Integration interval', options=('from slider', 'from keyboard'), value='from slider')
+      interval_s: IntRangeSlider(value=(0, 2), description=u'slider: ', max=10, min=-5)
+      interval_g: Grid(value=[[0, 2]], children=(Label(value=u'keyboard: '), VBox(children=(EvalText(value=u'0', layout=Layout(max_width=u'5em')),)), VBox(children=(EvalText(value=u'2', layout=Layout(max_width=u'5em')),))))
+      hr2: HTMLText(value=u'<hr>')
+      list_table: Checkbox(value=False, description=u'List table')
     <small>Adjust your data and click Update button. Click repeatedly for another random values.</small>
-    Riemann sum: \(\displaystyle\sum_{i=1}^{5} f(\eta_i)(x_i-x_{i-1})=...\)
-    Exact value of the integral \(\displaystyle\int_{0}^{2}x^{2} +
-    1\,\mathrm{d}x=4.666666666666668\)
+    Riemann sum: <script type="math/tex">\displaystyle\sum_{i=1}^{5} f(\eta_i)(x_i-x_{i-1})=...</script>
+    Exact value of the integral <script
+    type="math/tex">\displaystyle\int_{0}^{2}x^{2} +
+    1\,\mathrm{d}x=4.666666666666668</script>
 
     sage: test(interacts.calculus.function_tool)
     Interactive function <function function_tool at ...> with 7 widgets
-      f: EvalText(value='sin(x)', description='f')
-      g: EvalText(value='cos(x)', description='g')
-      xrange: IntRangeSlider(value=(0, 1), description='x-range', max=3, min=-3)
-      yrange: Text(value='auto', description='yrange')
-      a: IntSlider(value=1, description='a', max=3, min=-1)
-      action: ToggleButtons(description='h = ', options=('f', 'df/dx', 'int f', 'num f', 'den f', '1/f', 'finv', 'f+a', 'f-a', 'f*a', 'f/a', 'f^a', 'f(x+a)', 'f(x*a)', 'f+g', 'f-g', 'f*g', 'f/g', 'f(g)'), value='f')
-      do_plot: Checkbox(value=True, description='Draw Plots')
-    <center><font color="red">\(f = \sin\left(x\right)\)</font></center>
-    <center><font color="green">\(g = \cos\left(x\right)\)</font></center>
-    <center><font color="blue"><b>\(h = f = \sin\left(x\right)\)</b></font></center>
+      f: EvalText(value=u'sin(x)', description=u'f')
+      g: EvalText(value=u'cos(x)', description=u'g')
+      xrange: IntRangeSlider(value=(0, 1), description=u'x-range', max=3, min=-3)
+      yrange: Text(value=u'auto', description=u'yrange')
+      a: IntSlider(value=1, description=u'a', max=3, min=-1)
+      action: ToggleButtons(description=u'h = ', options=('f', 'df/dx', 'int f', 'num f', 'den f', '1/f', 'finv', 'f+a', 'f-a', 'f*a', 'f/a', 'f^a', 'f(x+a)', 'f(x*a)', 'f+g', 'f-g', 'f*g', 'f/g', 'f(g)'), value='f')
+      do_plot: Checkbox(value=True, description=u'Draw Plots')
+    <center><font color="red"><script type="math/tex">f = \sin\left(x\right)</script></font></center>
+    <center><font color="green"><script type="math/tex">g = \cos\left(x\right)</script></font></center>
+    <center><font color="blue"><b><script type="math/tex">h = f = \sin\left(x\right)</script></b></font></center>
 
     sage: test(interacts.fractals.mandelbrot)
     Interactive function <function mandelbrot at ...> with 6 widgets
-      expo: FloatSlider(value=2.0, description='expo', max=10.0, min=-10.0)
-      iterations: IntSlider(value=20, description='# iterations', min=1)
-      zoom_x: FloatRangeSlider(value=(-2.0, 1.0), description='Zoom X', max=2.0, min=-2.0, step=0.01)
-      zoom_y: FloatRangeSlider(value=(-1.5, 1.5), description='Zoom Y', max=2.0, min=-2.0, step=0.01)
-      plot_points: IntSlider(value=150, description='plot points', max=400, min=20, step=20)
-      dpi: IntSlider(value=80, description='dpi', max=200, min=20, step=10)
+      expo: FloatSlider(value=2.0, description=u'expo', max=10.0, min=-10.0)
+      iterations: IntSlider(value=20, description=u'# iterations', min=1)
+      zoom_x: FloatRangeSlider(value=(-2.0, 1.0), description=u'Zoom X', max=2.0, min=-2.0, step=0.01)
+      zoom_y: FloatRangeSlider(value=(-1.5, 1.5), description=u'Zoom Y', max=2.0, min=-2.0, step=0.01)
+      plot_points: IntSlider(value=150, description=u'plot points', max=400, min=20, step=20)
+      dpi: IntSlider(value=80, description=u'dpi', max=200, min=20, step=10)
     <h2>Mandelbrot Fractal</h2>
-    Recursive Formula: \(z \leftarrow z^{2.00} + c\) for \(c \in \mathbb{C}\)
+    Recursive Formula: <script type="math/tex">z \leftarrow z^{2.00} + c</script> for <script type="math/tex">c \in \mathbb{C}</script>
 
     sage: test(interacts.fractals.julia)
     Interactive function <function julia at ...> with 8 widgets
-      expo: FloatSlider(value=2.0, description='expo', max=10.0, min=-10.0)
-      c_real: FloatSlider(value=0.5, description='real part const.', max=2.0, min=-2.0, step=0.01)
-      c_imag: FloatSlider(value=0.5, description='imag part const.', max=2.0, min=-2.0, step=0.01)
-      iterations: IntSlider(value=20, description='# iterations', min=1)
-      zoom_x: FloatRangeSlider(value=(-1.5, 1.5), description='Zoom X', max=2.0, min=-2.0, step=0.01)
-      zoom_y: FloatRangeSlider(value=(-1.5, 1.5), description='Zoom Y', max=2.0, min=-2.0, step=0.01)
-      plot_points: IntSlider(value=150, description='plot points', max=400, min=20, step=20)
-      dpi: IntSlider(value=80, description='dpi', max=200, min=20, step=10)
+      expo: FloatSlider(value=2.0, description=u'expo', max=10.0, min=-10.0)
+      c_real: FloatSlider(value=0.5, description=u'real part const.', max=2.0, min=-2.0, step=0.01)
+      c_imag: FloatSlider(value=0.5, description=u'imag part const.', max=2.0, min=-2.0, step=0.01)
+      iterations: IntSlider(value=20, description=u'# iterations', min=1)
+      zoom_x: FloatRangeSlider(value=(-1.5, 1.5), description=u'Zoom X', max=2.0, min=-2.0, step=0.01)
+      zoom_y: FloatRangeSlider(value=(-1.5, 1.5), description=u'Zoom Y', max=2.0, min=-2.0, step=0.01)
+      plot_points: IntSlider(value=150, description=u'plot points', max=400, min=20, step=20)
+      dpi: IntSlider(value=80, description=u'dpi', max=200, min=20, step=10)
     <h2>Julia Fractal</h2>
-    Recursive Formula: \(z \leftarrow z^{2.00} + (0.50+0.50*\mathbb{I})\)
+    Recursive Formula: <script type="math/tex">z \leftarrow z^{2.00} + (0.50+0.50*\mathbb{I})</script>
 
     sage: test(interacts.fractals.cellular_automaton)
     Interactive function <function cellular_automaton at ...> with 3 widgets
-      N: IntSlider(value=100, description='Number of iterations', max=500, min=1)
-      rule_number: IntSlider(value=110, description='Rule number', max=255)
-      size: IntSlider(value=6, description='size of graphic', max=11, min=1)
+      N: IntSlider(value=100, description=u'Number of iterations', max=500, min=1)
+      rule_number: IntSlider(value=110, description=u'Rule number', max=255)
+      size: IntSlider(value=6, description=u'size of graphic', max=11, min=1)
     <h2>Cellular Automaton</h2><div style="white-space: normal;">"A cellular automaton is a collection of "colored" cells          on a grid of specified shape that evolves through a number of          discrete time steps according to a set of rules based on the          states of neighboring cells." &mdash;          <a target="_blank" href="http://mathworld.wolfram.com/CellularAutomaton.html">Mathworld,         Cellular Automaton</a></div>         <div>Rule 110 expands to 01110110</div>
 
     sage: test(interacts.geometry.unit_circle)
     Interactive function <function unit_circle at ...> with 2 widgets
-      function: Dropdown(description='function', options=(('sin(x)', 0), ('cos(x)', 1), ('tan(x)', 2)), value=0)
-      x: TransformFloatSlider(value=0.0, description='x', max=6.283185307179586, step=0.015707963267948967)
+      function: Dropdown(description=u'function', options=(('sin(x)', 0), ('cos(x)', 1), ('tan(x)', 2)), value=0)
+      x: TransformFloatSlider(value=0.0, description=u'x', max=6.283185307179586, step=0.015707963267948967)
     <div style="white-space: normal;">Lines of the same color have         the same length</div>
 
     sage: test(interacts.geometry.trigonometric_properties_triangle)
     Interactive function <function trigonometric_properties_triangle at ...> with 3 widgets
-      a0: IntSlider(value=30, description='A', max=360)
-      a1: IntSlider(value=180, description='B', max=360)
-      a2: IntSlider(value=300, description='C', max=360)
+      a0: IntSlider(value=30, description=u'A', max=360)
+      a1: IntSlider(value=180, description=u'B', max=360)
+      a2: IntSlider(value=300, description=u'C', max=360)
     <h2>Trigonometric Properties of a Triangle</h2>
-    \(\angle A = {60.000}^{\circ},\) \(\angle B = {45.000}^{\circ},\) \(\angle C = {75.000}^{\circ}\)
-    \(AB = 1.931852\), \(BC = 1.732051\), \(CA = 1.414214\)
-    Area of triangle \(ABC = 1.183013\)
+    <script type="math/tex">\angle A = {60.000}^{\circ},</script> <script type="math/tex">\angle B = {45.000}^{\circ},</script> <script type="math/tex">\angle C = {75.000}^{\circ}</script>
+    <script type="math/tex">AB = 1.931852</script>, <script type="math/tex">BC = 1.732051</script>, <script type="math/tex">CA = 1.414214</script>
+    Area of triangle <script type="math/tex">ABC = 1.183013</script>
 
     sage: test(interacts.geometry.special_points)
     Interactive function <function special_points at ...> with 10 widgets
-      title: HTMLText(value='<h2>Special points in triangle</h2>')
-      a0: IntSlider(value=30, description='A', max=360)
-      a1: IntSlider(value=180, description='B', max=360)
-      a2: IntSlider(value=300, description='C', max=360)
-      show_median: Checkbox(value=False, description='Medians')
-      show_pb: Checkbox(value=False, description='Perpendicular Bisectors')
-      show_alt: Checkbox(value=False, description='Altitudes')
-      show_ab: Checkbox(value=False, description='Angle Bisectors')
-      show_incircle: Checkbox(value=False, description='Incircle')
+      title: HTMLText(value=u'<h2>Special points in triangle</h2>')
+      a0: IntSlider(value=30, description=u'A', max=360)
+      a1: IntSlider(value=180, description=u'B', max=360)
+      a2: IntSlider(value=300, description=u'C', max=360)
+      show_median: Checkbox(value=False, description=u'Medians')
+      show_pb: Checkbox(value=False, description=u'Perpendicular Bisectors')
+      show_alt: Checkbox(value=False, description=u'Altitudes')
+      show_ab: Checkbox(value=False, description=u'Angle Bisectors')
+      show_incircle: Checkbox(value=False, description=u'Incircle')
       show_euler: Checkbox(value=False, description="Euler's Line")
 
     sage: test(interacts.statistics.coin)
     Interactive function <function coin at ...> with 2 widgets
-      n: IntSlider(value=1000, description='Number of Tosses', max=10000, min=2, step=100)
-      interval: IntRangeSlider(value=(0, 0), description='Plotting range (y)', max=1)
+      n: IntSlider(value=1000, description=u'Number of Tosses', max=10000, min=2, step=100)
+      interval: IntRangeSlider(value=(0, 0), description=u'Plotting range (y)', max=1)
     doctest:...: UserWarning: Attempting to set identical bottom == top == 0.0 results in singular transformations; automatically expanding.
 
 Test matrix control (see :trac:`27735`)::
@@ -290,7 +291,7 @@
     ....:     print(parent(A))
     sage: test(matrix_test)
     Interactive function <function matrix_test at ...> with 1 widget
-      A: Grid(value=[[0, 1], [2, 3]], children=(Label(value='A'), VBox(children=(EvalText(value='0', layout=Layout(max_width='5em')), EvalText(value='2', layout=Layout(max_width='5em')))), VBox(children=(EvalText(value='1', layout=Layout(max_width='5em')), EvalText(value='3', layout=Layout(max_width='5em'))))))
+      A: Grid(value=[[0, 1], [2, 3]], children=(Label(value=u'A'), VBox(children=(EvalText(value=u'0', layout=Layout(max_width=u'5em')), EvalText(value=u'2', layout=Layout(max_width=u'5em')))), VBox(children=(EvalText(value=u'1', layout=Layout(max_width=u'5em')), EvalText(value=u'3', layout=Layout(max_width=u'5em'))))))
     [0 1]
     [2 3]
     Full MatrixSpace of 2 by 2 dense matrices over Rational Field
--- a/sage/src/sage/interfaces/mathematica.py
+++ b/sage/src/sage/interfaces/mathematica.py
@@ -232,7 +232,7 @@
     sage: F[4]                               # optional - mathematica
     {541, 1}
 
-Mathematica's ECM package is no longer available.
+Mathematica's ECM package is no longer available. 
 
 Long Input
 ----------
--- a/sage/src/sage/misc/html.py
+++ b/sage/src/sage/misc/html.py
@@ -64,8 +64,10 @@
 
 def math_parse(s):
     r"""
-    Transform the string ``s`` with TeX maths to an HTML string renderable by
-    MathJax.
+    Replace TeX-``$`` with Mathjax equations.
+
+    Turn the HTML-ish string s that can have \$\$ and \$'s in it into
+    pure HTML.  See below for a precise definition of what this means.
 
     INPUT:
 
@@ -75,25 +77,48 @@
 
     A :class:`HtmlFragment` instance.
 
-    Specifically this method does the following:
+    Do the following:
 
-    * Replace all ``\$text\$``\'s by ``\(text\)``
-    * Replace all ``\$\$text\$\$``\'s by ``\[text\]``
-    * Replace all ``\\\$``\'s by ``\$``\'s. Note that this has precedence over
-      the above two cases.
+    * Replace all ``\$ text \$``\'s by
+      ``<script type="math/tex"> text </script>``
+    * Replace all ``\$\$ text \$\$``\'s by
+      ``<script type="math/tex; mode=display"> text </script>``
+    * Replace all ``\ \$``\'s by ``\$``\'s.  Note that in
+      the above two cases nothing is done if the ``\$``
+      is preceeded by a backslash.
+    * Replace all ``\[ text \]``\'s by
+      ``<script type="math/tex; mode=display"> text </script>``
 
     EXAMPLES::
 
-        sage: print(sage.misc.html.math_parse('This is $2+2$.'))
-        This is \(2+2\).
-        sage: print(sage.misc.html.math_parse('This is $$2+2$$.'))
-        This is \[2+2\].
-        sage: print(sage.misc.html.math_parse('This is \\[2+2\\].'))
-        This is \[2+2\].
-        sage: print(sage.misc.html.math_parse(r'\$2+2\$ is rendered to $2+2$.'))
-        <span>$</span>2+2<span>$</span> is rendered to \(2+2\).
+        sage: pretty_print(sage.misc.html.math_parse('This is $2+2$.'))
+        This is <script type="math/tex">2+2</script>.
+        sage: pretty_print(sage.misc.html.math_parse('This is $$2+2$$.'))
+        This is <script type="math/tex; mode=display">2+2</script>.
+        sage: pretty_print(sage.misc.html.math_parse('This is \\[2+2\\].'))
+        This is <script type="math/tex; mode=display">2+2</script>.
+        sage: pretty_print(sage.misc.html.math_parse(r'This is \[2+2\].'))
+        This is <script type="math/tex; mode=display">2+2</script>.
+
+    TESTS::
 
+        sage: sage.misc.html.math_parse(r'This \$\$is $2+2$.')
+        This $$is <script type="math/tex">2+2</script>.
     """
+    # first replace \\[ and \\] by <script type="math/tex; mode=display">
+    # and </script>, respectively.
+    while True:
+        i = s.find('\\[')
+        if i == -1:
+            break
+        else:
+            s = s[:i] + '<script type="math/tex; mode=display">' + s[i+2:]
+            j = s.find('\\]')
+            if j == -1:  # missing right-hand delimiter, so add one
+                s = s + '</script>'
+            else:
+                s = s[:j] + '</script>' + s[j+2:]
+
     # Below t always has the "parsed so far" version of s, and s is
     # just the part of the original input s that hasn't been parsed.
     t = ''
@@ -103,19 +128,18 @@
             # No dollar signs -- definitely done.
             return HtmlFragment(t + s)
         elif i > 0 and s[i-1] == '\\':
-            # A dollar sign with a backslash right before it, so this is a
-            # normal dollar sign. If processEscapes is enabled in MathJax, "\$"
-            # will do the job. But as we do not assume that, we use the span
-            # tag safely.
-            t += s[:i-1] + '<span>$</span>'
+            # A dollar sign with a backslash right before it, so
+            # we ignore it by sticking it in the parsed string t
+            # and skip to the next iteration.
+            t += s[:i-1] + '$'
             s = s[i+1:]
             continue
         elif i+1 < len(s) and s[i+1] == '$':
             # Found a math environment. Double dollar sign so display mode.
-            disp = True
+            disp = '; mode=display'
         else:
             # Found math environment. Single dollar sign so default mode.
-            disp = False
+            disp = ''
 
         # Now find the matching $ sign and form the html string.
 
@@ -136,10 +160,8 @@
                 j += i + 2
             txt = s[i+1:j]
 
-        if disp:
-            t += s[:i] + r'\[{0}\]'.format(' '.join(txt.splitlines()))
-        else:
-            t += s[:i] + r'\({0}\)'.format(' '.join(txt.splitlines()))
+        t += s[:i] + '<script type="math/tex%s">%s</script>' % (disp,
+                      ' '.join(txt.splitlines()))
         s = s[j+1:]
         if disp:
             s = s[1:]
@@ -227,9 +249,9 @@
 
         sage: from sage.misc.html import MathJax
         sage: MathJax()(3)
-        <html>\[\newcommand{\Bold}[1]{\mathbf{#1}}3\]</html>
+        <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbf{#1}}3</script></html>
         sage: MathJax()(ZZ)
-        <html>\[\newcommand{\Bold}[1]{\mathbf{#1}}\Bold{Z}\]</html>
+        <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbf{#1}}\Bold{Z}</script></html>
     """
 
     def __call__(self, x, combine_all=False):
@@ -253,7 +275,7 @@
 
             sage: from sage.misc.html import MathJax
             sage: MathJax()(3)
-            <html>\[\newcommand{\Bold}[1]{\mathbf{#1}}3\]</html>
+            <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbf{#1}}3</script></html>
             sage: str(MathJax().eval(ZZ['x'], mode='display')) == str(MathJax()(ZZ['x']))
             True
         """
@@ -290,7 +312,7 @@
 
             sage: from sage.misc.html import MathJax
             sage: MathJax().eval(3, mode='display')
-            <html>\[\newcommand{\Bold}[1]{\mathbf{#1}}3\]</html>
+            <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbf{#1}}3</script></html>
             sage: MathJax().eval(3, mode='inline')
             <html>\(\newcommand{\Bold}[1]{\mathbf{#1}}3\)</html>
             sage: MathJax().eval(type(3), mode='inline')
@@ -351,9 +373,9 @@
         )
         mathjax_string = latex_string.replace('<', '&lt;')
         if mode == 'display':
-            html = r'<html>\[{0}\]</html>'
+            html = '<html><script type="math/tex; mode=display">{0}</script></html>'
         elif mode == 'inline':
-            html = r'<html>\({0}\)</html>'
+            html = '<html><script type="math/tex">{0}</script></html>'
         elif mode == 'plain':
             return mathjax_string
         else:
@@ -405,7 +427,7 @@
             <class 'sage.misc.html.HtmlFragment'>
 
             sage: html(1/2)
-            <html>\[\newcommand{\Bold}[1]{\mathbf{#1}}\frac{1}{2}\]</html>
+            <html><script type="math/tex; mode=display">\newcommand{\Bold}[1]{\mathbf{#1}}\frac{1}{2}</script></html>
 
             sage: html('<a href="http://sagemath.org">sagemath</a>')
             <a href="http://sagemath.org">sagemath</a>
@@ -452,9 +474,9 @@
 
             sage: a = 123
             sage: html.eval('<sage>a</sage>')
-            \(123\)
+            <script type="math/tex">123</script>
             sage: html.eval('<sage>a</sage>', locals={'a': 456})
-            \(456\)
+            <script type="math/tex">456</script>
         """
         if locals is None:
             from sage.repl.user_globals import get_globals
@@ -471,7 +493,8 @@
             if j == -1:
                  t += s
                  break
-            t += s[:i] + r'\({}\)'.format(latex(sage_eval(s[6+i:j], locals=locals)))
+            t += s[:i] + '<script type="math/tex">%s</script>'%\
+                     latex(sage_eval(s[6+i:j], locals=locals))
             s = s[j+7:]
         return HtmlFragment(t)
 
--- a/sage/src/sage/misc/table.py
+++ b/sage/src/sage/misc/table.py
@@ -179,24 +179,24 @@
         <table border="1" class="table_form">
         <tbody>
         <tr>
-        <th style="text-align:left">\(x\)</th>
-        <th style="text-align:left">\(\sin(x)\)</th>
+        <th><script type="math/tex">x</script></th>
+        <th><script type="math/tex">\sin(x)</script></th>
         </tr>
         <tr class ="row-a">
-        <td style="text-align:left">\(0\)</td>
-        <td style="text-align:left">\(0.00\)</td>
+        <td><script type="math/tex">0</script></td>
+        <td><script type="math/tex">0.00</script></td>
         </tr>
         <tr class ="row-b">
-        <td style="text-align:left">\(1\)</td>
-        <td style="text-align:left">\(0.84\)</td>
+        <td><script type="math/tex">1</script></td>
+        <td><script type="math/tex">0.84</script></td>
         </tr>
         <tr class ="row-a">
-        <td style="text-align:left">\(2\)</td>
-        <td style="text-align:left">\(0.91\)</td>
+        <td><script type="math/tex">2</script></td>
+        <td><script type="math/tex">0.91</script></td>
         </tr>
         <tr class ="row-b">
-        <td style="text-align:left">\(3\)</td>
-        <td style="text-align:left">\(0.14\)</td>
+        <td><script type="math/tex">3</script></td>
+        <td><script type="math/tex">0.14</script></td>
         </tr>
         </tbody>
         </table>
@@ -648,22 +648,22 @@
             <table  class="table_form">
             <tbody>
             <tr class ="row-a">
-            <td style="text-align:left">\(\sin(x)\)</td>
-            <td style="text-align:left">\(x\)</td>
-            <td style="text-align:left">text</td>
+            <td><script type="math/tex">\sin(x)</script></td>
+            <td><script type="math/tex">x</script></td>
+            <td>text</td>
             </tr>
             <tr class ="row-b">
-            <td style="text-align:left">\(1\)</td>
-            <td style="text-align:left">\(34342\)</td>
-            <td style="text-align:left">\(3\)</td>
+            <td><script type="math/tex">1</script></td>
+            <td><script type="math/tex">34342</script></td>
+            <td><script type="math/tex">3</script></td>
             </tr>
             <tr class ="row-a">
-            <td style="text-align:left">\(\left(\begin{array}{rr}
+            <td><script type="math/tex">\left(\begin{array}{rr}
             1 & 0 \\
             0 & 1
-            \end{array}\right)\)</td>
-            <td style="text-align:left">\(5\)</td>
-            <td style="text-align:left">\(6\)</td>
+            \end{array}\right)</script></td>
+            <td><script type="math/tex">5</script></td>
+            <td><script type="math/tex">6</script></td>
             </tr>
             </tbody>
             </table>
@@ -690,24 +690,24 @@
             <table border="1" class="table_form">
             <tbody>
             <tr>
-            <th style="text-align:left">\(x\)</th>
-            <th style="text-align:left">\(\sin(x)\)</th>
+            <th><script type="math/tex">x</script></th>
+            <th><script type="math/tex">\sin(x)</script></th>
             </tr>
             <tr class ="row-a">
-            <td style="text-align:left">\(0\)</td>
-            <td style="text-align:left">\(0.00\)</td>
+            <td><script type="math/tex">0</script></td>
+            <td><script type="math/tex">0.00</script></td>
             </tr>
             <tr class ="row-b">
-            <td style="text-align:left">\(1\)</td>
-            <td style="text-align:left">\(0.84\)</td>
+            <td><script type="math/tex">1</script></td>
+            <td><script type="math/tex">0.84</script></td>
             </tr>
             <tr class ="row-a">
-            <td style="text-align:left">\(2\)</td>
-            <td style="text-align:left">\(0.91\)</td>
+            <td><script type="math/tex">2</script></td>
+            <td><script type="math/tex">0.91</script></td>
             </tr>
             <tr class ="row-b">
-            <td style="text-align:left">\(3\)</td>
-            <td style="text-align:left">\(0.14\)</td>
+            <td><script type="math/tex">3</script></td>
+            <td><script type="math/tex">0.14</script></td>
             </tr>
             </tbody>
             </table>
@@ -776,9 +776,9 @@
             sage: s = StringIO()
             sage: T._html_table_row(s, ['a', 2, '$x$'])
             sage: print(s.getvalue())
-            <td style="text-align:left">a</td>
-            <td style="text-align:left">\(2\)</td>
-            <td style="text-align:left">\(x\)</td>
+            <td>a</td>
+            <td><script type="math/tex">2</script></td>
+            <td><script type="math/tex">x</script></td>
         """
         from sage.plot.all import Graphics
         from .latex import latex
@@ -790,40 +790,27 @@
         elif not isinstance(row, (list, tuple)):
             row = [row]
 
-        align_char = self._options['align'][0]   # 'l', 'c', 'r'
-
-        if align_char == 'l':
-            style = 'text-align:left'
-        elif align_char == 'c':
-            style = 'text-align:center'
-        elif align_char == 'r':
-            style = 'text-align:right'
-        else:
-            style = ''
-
-        style_attr = f' style="{style}"' if style else ''
-
-        column_tag = f'<th{style_attr}>%s</th>\n' if header else f'<td{style_attr}>%s</td>\n'
+        column_tag = "<th>%s</th>\n" if header else "<td>%s</td>\n"
 
         if self._options['header_column']:
-            first_column_tag = '<th class="ch"{style_attr}>%s</th>\n' if header else '<td class="ch"{style_attr}>%s</td>\n'
+            first_column_tag = '<th class="ch">%s</th>\n' if header else '<td class="ch">%s</td>\n'
         else:
             first_column_tag = column_tag
 
-        # first entry of row
+        # First entry of row:
         entry = row[0]
         if isinstance(entry, Graphics):
             file.write(first_column_tag % entry.show(linkmode = True))
         elif isinstance(entry, str):
             file.write(first_column_tag % math_parse(entry))
         else:
-            file.write(first_column_tag % (r'\(%s\)' % latex(entry)))
+            file.write(first_column_tag % ('<script type="math/tex">%s</script>' % latex(entry)))
 
-        # other entries
+        # Other entries:
         for column in range(1, len(row)):
             if isinstance(row[column], Graphics):
                 file.write(column_tag % row[column].show(linkmode = True))
             elif isinstance(row[column], str):
                 file.write(column_tag % math_parse(row[column]))
             else:
-                file.write(column_tag % (r'\(%s\)' % latex(row[column])))
+                file.write(column_tag % ('<script type="math/tex">%s</script>' % latex(row[column])))
--- a/sage/src/sage/repl/rich_output/backend_base.py
+++ b/sage/src/sage/repl/rich_output/backend_base.py
@@ -454,23 +454,23 @@
             sage: out
             OutputHtml container
             sage: out.html
-            buffer containing 62 bytes
+            buffer containing 105 bytes
             sage: out.html.get_str()
-            '<html>\\[\\newcommand{\\Bold}[1]{\\mathbf{#1}}\\frac{1}{2}\\]</html>'
+            '<html><script type="math/tex; mode=display">\\newcommand{\\Bold}[1]{\\mathbf{#1}}\\frac{1}{2}</script></html>'
 
             sage: out = backend.latex_formatter([1/2, x, 3/4, ZZ], concatenate=False)
             sage: out.html.get_str()
-            '<html>\\[\\newcommand{\\Bold}[1]{\\mathbf{#1}}\\left[\\frac{1}{2}, x, \\frac{3}{4}, \\Bold{Z}\\right]\\]</html>'
+            '<html><script type="math/tex; mode=display">\\newcommand{\\Bold}[1]{\\mathbf{#1}}\\left[\\frac{1}{2}, x, \\frac{3}{4}, \\Bold{Z}\\right]</script></html>'
             sage: out = backend.latex_formatter([1/2, x, 3/4, ZZ], concatenate=True)
             sage: out.html.get_str()
-            '<html>\\[\\newcommand{\\Bold}[1]{\\mathbf{#1}}\\frac{1}{2} x \\frac{3}{4} \\Bold{Z}\\]</html>'
+            '<html><script type="math/tex; mode=display">\\newcommand{\\Bold}[1]{\\mathbf{#1}}\\frac{1}{2} x \\frac{3}{4} \\Bold{Z}</script></html>'
 
         TESTS::
 
             sage: backend.latex_formatter([], concatenate=False).html.get_str()
-            '<html>\\[\\newcommand{\\Bold}[1]{\\mathbf{#1}}\\left[\\right]\\]</html>'
+            '<html><script type="math/tex; mode=display">\\newcommand{\\Bold}[1]{\\mathbf{#1}}\\left[\\right]</script></html>'
             sage: backend.latex_formatter([], concatenate=True).html.get_str()
-            '<html>\\[\\newcommand{\\Bold}[1]{\\mathbf{#1}}\\]</html>'
+            '<html><script type="math/tex; mode=display">\\newcommand{\\Bold}[1]{\\mathbf{#1}}</script></html>'
         """
         concatenate = kwds.get('concatenate', False)
         from sage.misc.html import html
--- a/sage/src/sage/repl/rich_output/output_browser.py
+++ b/sage/src/sage/repl/rich_output/output_browser.py
@@ -8,9 +8,8 @@
 from sage.repl.rich_output.output_basic import OutputBase
 from sage.repl.rich_output.buffer import OutputBuffer
 
-# regex to match "<html>\[...\]</html>" or "<html>\(...\)</html>"
-latex_re = re.compile(r'<html>(?P<mathstart>\\\[|\\\()(?P<latex>.*)(?P<mathend>\\\]|\\\))</html>',
-                      flags=re.DOTALL)
+latex_re = re.compile(r'<html><script type="math/tex; mode=(?P<mode>[^"]+)">(?P<latex>.*)</script></html>',
+                     flags=re.DOTALL)
 
 class OutputHtml(OutputBase):
 
@@ -21,9 +20,10 @@
         INPUT:
 
         - ``html`` --
-          :class:`~sage.repl.rich_output.buffer.OutputBuffer`. Alternatively, a
-          string (bytes) can be passed directly which will then be converted
-          into an :class:`~sage.repl.rich_output.buffer.OutputBuffer`. String
+          :class:`~sage.repl.rich_output.buffer.OutputBuffer`. Alternatively,
+          a string (bytes) can be passed directly which will then be
+          converted into an
+          :class:`~sage.repl.rich_output.buffer.OutputBuffer`. String
           containing the html fragment code. Excludes the surrounding
           ``<body>`` and ``<html>`` tag.
 
@@ -42,7 +42,7 @@
         if m:
             mathjax_string = m.group('latex')
             latex_string = mathjax_string.replace('&lt;', '<')
-            if m.group('mathstart') == r'\[' and m.group('mathend') == r'\]':
+            if m.group('mode') == 'display':
                 self.latex = OutputBuffer('$$' + latex_string + '$$')
             else:
                 self.latex = OutputBuffer('$' + latex_string + '$')
--- a/sage/src/sage/repl/rich_output/pretty_print.py
+++ b/sage/src/sage/repl/rich_output/pretty_print.py
@@ -289,9 +289,9 @@
 
         sage: plt = plot(sin)
         sage: pretty_print(plt)             # graphics output
+        sage: pretty_print(ZZ, 123, plt)    # optional - latex
+        <html><script type="math/tex">\newcommand{\Bold}[1]{\mathbf{#1}}\Bold{Z} 123 %% Creator: Matplotlib, PGF backend...</script></html>
         sage: pretty_print(plt, plt)        # graphics output
-        sage: pretty_print(ZZ, 123, plt)
-        Integer Ring 123 Graphics object consisting of 1 graphics primitive
     """
     dm = get_display_manager()
     old_preferences_text = dm.preferences.text
